src/pages/MyCoursePage.tsx

```diff
@@ -0,0 +1,327 @@
+                        weight: 40,
+                        type: 'Mündliche Prüfung',
+                    },
+                    {
+                        id: 5,
+                        description: 'Test #2',
+                        weight: 20,
+                        type: 'Mündliche Prüfung',
+                    },
+                    {
+                        id: 6,
+                        description: 'Test #3',
+                        weight: 40,
+                        type: 'Schriftliche Prüfung',
+                    },
+                ],
+                teachers: [
+                    {
+                        id: 2,
+                        name: 'John Doe',
+                    },
+                ],
+            },
+            {
+                id: 3,
+                description: 'Software-Architektur',
+                exams: [
+                    {
+                        id: 7,
+                        description: 'Test #1',
+                        weight: 50,
+                        type: 'Schriftliche Prüfung',
+                    },
+                    {
+                        id: 8,
+                        description: 'Test #2',
+                        weight: 30,
+                        type: 'Mündliche Prüfung',
+                    },
+                    {
+                        id: 9,
+                        description: 'Test #3',
+                        weight: 20,
+                        type: 'Mündliche Prüfung',
+                    },
+                ],
+                teachers: [
+                    {
+                        id: 2,
+                        name: 'John Doe',
+                    },
+                ],
+            },
+        ],
+    },
+];
+
+export default function MyCoursePage() {
+    const dropdownItems = customCourses.map((customCourse, index) => ({
+        label: customCourse.startDate,
+        value: index,
+    }));
+
+    type MenuItem = Required<MenuProps>['items'][number];
+
+    const getItem = React.useCallback(
+        (
+            label: React.ReactNode,
+            key: React.Key,
+            icon?: React.ReactNode,
+            children?: MenuItem[],
+            type?: 'group'
+        ): MenuItem => {
+            return {
+                key,
+                icon,
+                children,
+                label,
+                type,
+            } as MenuItem;
+        },
+        []
+    );
+
+    const getMenuItems = React.useCallback(
+        (key: number) => {
+            return customCourses[key].course.map((course, index) =>
+                getItem(course.description, index.toString(), <BookFilled />)
+            );
+        },
+        [getItem]
+    );
+
+    const intl = useIntl();
+    const [menuItems, setMenuItems] = useState<MenuItem[]>(getMenuItems(0));
+    const [current, setCurrent] = useState<MenuItemType>(menuItems[0] as MenuItemType);
+
+    const handleOnChange = React.useCallback(
```

Comment:
just use `useCallback()` directly and add it to the imports like you did for useEffect and useState